repository/main.tf

```diff
@@ -0,0 +1,89 @@
+resource "aws_ecr_repository" "ecr-repo" {
+  name = "${var.repository_name}"
+}
+
+resource "aws_iam_policy" "ecr-repo-push" {
+  name        = "ecr-${var.repository_name}-push"
+  path        = "/"
+  description = "Push access to the ${var.repository_description} repository"
+
+  policy = <<EOF
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Action": [
+        "ecr:DescribeRepositories",
+        "ecr:ListImages",
+        "ecr:DescribeRepositories",
+        "ecr:ListImages",
+        "ecr:BatchCheckLayerAvailability",
+        "ecr:BatchGetImage",
+        "ecr:DescribeImages",
+        "ecr:GetAuthorizationToken",
+        "ecr:GetDownloadUrlForLayer",
+        "ecr:GetRepositoryPolicy",
+        "ecr:CompleteLayerUpload",
+        "ecr:InitiateLayerUpload",
+        "ecr:PutImage",
+        "ecr:UploadLayerPart"
+      ],
+      "Effect": "Allow",
+      "Resource": "${aws_ecr_repository.ecr-repo.arn}"
+    }
+  ]
+}
+EOF
+}
+
+resource "aws_iam_policy" "ecr-repo-pull" {
+  name        = "ecr-${var.repository_name}-pull"
+  path        = "/"
+  description = "Pull access to the ${var.repository_description} repository"
+
+  policy = <<EOF
```

Comment:
Can you use this to specify the policy?
 https://www.terraform.io/docs/providers/aws/d/iam_policy_document.html

Also I'd set the action as variable so that it can be changed if needed
And I'd also give the option to create multiple repos

Reply:
@venturel multiple repos? How would you specify that? I am targeting this module to have really tight ACL around repo access.